### DISCUSSION TITLE
如何自定义映射实体类型

### DISCUSSION THREAD

**Discussion by ekirio** [dotnetcore/FreeSql#1185] [Q&A]
在postgresql上使用jsonmap发现不能工作,求助如何自定义映射实体类型.
例如:数据库配的varchar, 如何在查询时自动映射IEnumerable类型,
类似AOP中AuditValue在新增和更新时映射,查询时该如何实现

**Comment by 2881099** [comment]
JsonMap 尽量使用独立的 class 映射，避免冲突

class JsonEnumable
{
    public IEumable\<string> xxx { get; set; }
}

**Comment by ekirio** [comment]
尝试使用独立的类后在使用CodeFirst情况下不会生成对应数据库结构,手动创建结构后, 新增数据对应的jsonmap列还是没法添加到表中.
先尝试手动序列解决了
